src/engine/sequencer.js

```diff
@@ -131,12 +131,17 @@ Sequencer.prototype.stepThread = function (thread) {
         // If no next block has been found at this point, look on the stack.
         while (!thread.peekStack()) {
             thread.popStack();
+
             if (thread.stack.length === 0) {
                 // No more stack to run!
                 thread.status = Thread.STATUS_DONE;
                 return;
             }
-            if (thread.peekStackFrame().isLoop) {
+			
+			var stackFrame = thread.peekStackFrame();
+	        isWarpMode = stackFrame.warpMode;
```

Comment:
GitHub and your text editor aren't working together.. the indentation here is a little messed up. Should be 12 spaces.